### PR TITLE
feat: Add image content section to images page

### PR DESCRIPTION
**Pull request by sergeyshevch**
Hello! This PR is aiming to solve the remaining items in #4502 especially addressing https://github.com/kubernetes/website/issues/4502#issuecomment-570288311

### CODE REVIEW

**Review by thisisharrsh — COMMENTED** [review]
How about we make a tooltip for `OCI artefact` so that readers can get more information about OCI?

**Comment by thisisharrsh** [comment]
Deployment preview : https://deploy-preview-46440--kubernetes-io-main-staging.netlify.app/docs/concepts/containers/images/

File: content/en/docs/concepts/containers/images.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ hide_summary: true # Listed separately in section index
 
 <!-- overview -->
 
-A container image represents binary data that encapsulates an application and all its
+A container image is an OCI artifact that that encapsulates an application and all its
```
Comment (sftim) [inline content/en/docs/concepts/containers/images.md:13]: This does need either a glossary tooltip or a plain hyperlink. Without it, we're making the page more confusing, not less.

File: content/en/docs/concepts/containers/images.md [hunk 2]
```diff
@@ -63,6 +63,21 @@ Some image name examples that Kubernetes can use are:
 - `registry.k8s.io/pause@sha256:1ff6c18fbef2045af6b9c16bf034cc421a29027b800e4f9b68ae9b1cb3e9ae07` - Image name with digest.
 - `registry.k8s.io/pause:3.5@sha256:1ff6c18fbef2045af6b9c16bf034cc421a29027b800e4f9b68ae9b1cb3e9ae07` - Image name with tag and digest. Only digest will be used for pulling.
 
+## Image content
+
+OCI images are composed of layers. Each layer is a set of files that are added to the image.
```
Comment (sftim) [inline content/en/docs/concepts/containers/images.md:68]:
```suggestion
Container images are composed of _layers_. Each layer is a set of files that are added to the image.
```

Docker-format images have layers too. It'd be wrong to imply that only OCI images have layers.

File: content/en/docs/concepts/containers/images.md [hunk 3]
```diff
@@ -63,6 +63,21 @@ Some image name examples that Kubernetes can use are:
...
+
+OCI Container images can be created using various tools and techniques.
+Some examples are: Docker, Buildah, Kaniko, and Buildpacks.
+
+For more information on the OCI Image Specification, see the [OCI Image Spec](https://github.com/opencontainers/image-spec).
+
+{{< note >}}
+Other types of artifacts that are OCI compliant but not container images 
+cannot be used by Kubernetes as images. You can read more about them in the [OCI Manifest Spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md).
+{{< /note >}}
```
Comment (sftim) [inline content/en/docs/concepts/containers/images.md:79]: Why is this important to mention? I wouldn't use a note callout here.

Try this:

```suggestion
### Artifacts that are _not_ images {#other-oci-artifacts}

Container images are only one kind of OCI artifact. You can read more about them in the [OCI Manifest Spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md). Types of artifacts that are OCI compliant but not container images cannot be used by Kubernetes as container images.
```

File: content/en/docs/concepts/containers/images.md [hunk 4]
```diff
@@ -63,6 +63,21 @@ Some image name examples that Kubernetes can use are:
 - `registry.k8s.io/pause@sha256:1ff6c18fbef2045af6b9c16bf034cc421a29027b800e4f9b68ae9b1cb3e9ae07` - Image name with digest.
 - `registry.k8s.io/pause:3.5@sha256:1ff6c18fbef2045af6b9c16bf034cc421a29027b800e4f9b68ae9b1cb3e9ae07` - Image name with tag and digest. Only digest will be used for pulling.
 
+## Image content
+
+OCI images are composed of layers. Each layer is a set of files that are added to the image.
+Each layer and final image are identified by a digest. The digest of layer is a hash of the layer's content.
+
+OCI Container images can be created using various tools and techniques.
```
Comment (sftim) [inline content/en/docs/concepts/containers/images.md:71]:
```suggestion
Container images can be created using various tools and techniques.
```

File: content/en/docs/concepts/containers/images.md [hunk 5]
```diff
@@ -63,6 +63,21 @@ Some image name examples that Kubernetes can use are:
...
+Some examples are: Docker, Buildah, Kaniko, and Buildpacks.
```
Comment (sftim) [inline content/en/docs/concepts/containers/images.md:72]: - Alphabetical order please
- And use a shortcode marker - see https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#third-party-content-item

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks. With tweaks this could merge; without them, I need to request changes.

**Comment by sftim** [comment]
@sergeyshevch: do you have the time to address the feedback so far?

**Pull request closed without merging**